src/server/dfly_main.cc

```diff
@@ -438,6 +439,114 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
+string get_cgroup_path(void) {
+  // Begin by reading /proc/self/cgroup
+
+  auto cg = io::ReadFileToString("/proc/self/cgroup");
+  CHECK(cg.has_value());
+
+  // Here, we assume that CGroup v1 is being used. This
+  // is quite likely, as CGroup v1 was introduced back in 2015.
+
+  // strip 0::<path> into just <path>, and newline.
+  auto cgv = std::move(cg.value());
```

Comment:
Please avoid 100% "auto" usage - it hurts readability. In this specific case I would replace "auto" with "string"